src/common_dims.cpp

```diff
@@ -152,5 +146,42 @@ common_dims common_dims::compute(const std::vector<std::size_t>& dims1,
     return cd;
 }
 
+const std::vector<std::vector<std::size_t>>* common_dims::get_axes_map(std::size_t n) const
+{
+    if(axes_map1.size() == n)
+        return &axes_map1;
+    if(axes_map2.size() == n)
+        return &axes_map2;
+    return nullptr;
+}
+
+std::vector<std::size_t>
+common_dims::get_dimensions_for(const std::vector<std::size_t>& idims) const
+{
+    if(elements(dims) == elements(idims))
+        return dims;
+    auto* axes_map = get_axes_map(idims.size());
+    if(axes_map == nullptr)
+        return {};
+    auto xdims = dims;
+    for(auto i : range(axes_map->size()))
+    {
+        auto dim         = idims[i];
+        const auto& axes = (*axes_map)[i];
+        if(axes.size() == 1)
+        {
+            xdims[axes.front()] = dim;
+        }
+        else if(dim == 1)
+        {
+            for(auto axis : axes)
+                xdims[axis] = 1;
```

Comment:
I think setting it as 1 would ensure `elements(xdims) != elements(idims)`.

Reply:
No it wont. So say an axis of 2 is mapped to axes of `{3, 4}`, then if the dimension is `1` then the number of elements in axes `{3, 4}` should total `1`, so we can easily set it to `1`. If the dimension wasn't 1 and something like 8 then there are several combinations(ie `{1, 8}`, `{4, 2}`, `{8, 1}`, `{2, 4}`, etc), so that is why we dont update it.